activerecord/test/cases/serialized_attribute_test.rb

```diff
@@ -403,6 +403,27 @@ def test_values_cast_from_nil_are_persisted_as_nil
     assert_equal [topic, topic2], Topic.where(content: nil).sort_by(&:id)
   end
 
+  # MySQL doesn't support default values for text columns, so we need to skip this test for MySQL
+  if !current_adapter?(:Mysql2Adapter)
+    def test_serialized_attribute_with_default_can_update_to_default
+      ActiveRecord::Schema.define do
+        create_table :tmp_posts, force: true do |t|
+          t.text     :content, null: false, default: "{}"
+        end
+      end
+      klass = Class.new(ActiveRecord::Base) do
+        self.table_name = "tmp_posts"
+        serialize(:content, Hash)
```

Comment:
I'd like a test with a serializer that returns `nil`

Reply:
@byroot please correct me if I'm wrong, but original serializer doesn't return `nil`, it returns blank `Hash`, that it the root cause. But we don't want to fix that, as many people may rely on this behaviour.
And we have tests above which test the behaviour of trying to update to `nil`

Reply:
> And we have tests above which test the behaviour of trying to update to nil

But in these tests the columns doesn't have a default value, does it?

Reply:
@byroot addressed in the new test, similar to another test (where default is not specified). Updating here from `Hash` to `nil` leads to some unexpected behaviour which I mentioned above.

Reply:
Thanks. Please squash your commits.

Reply:
@byroot Done. Thank you for your patience, feedback and support!